src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -764,48 +783,75 @@ void SettingsLoader::_parseFragment(const winrt::hstring& source, const std::str
         {
             try
             {
-                auto profile = _parseProfile(OriginTag::Fragment, source, profileJson);
                 // GH#9962: Discard Guid-less, Name-less profiles, but...
                 // allow ones with an Updates field, as those are special for fragments.
                 // We need to make sure to only call Guid() if HasGuid() is true,
                 // as Guid() will dynamically generate a return value otherwise.
-                const auto guid = profile->HasGuid() ? profile->Guid() : profile->Updates();
-                if (guid != winrt::guid{})
+                auto profile = _parseProfile(OriginTag::Fragment, source, profileJson);
+                if (const auto guid = profile->Guid(); profile->HasGuid() && guid != winrt::guid{})

```

Comment:
```suggestion
                const auto guid = profile->HasGuid() ? profile->Guid() : profile->Updates();
                const auto destinationSet = profile->HasGuid() ? fragmentSettings->NewProfiles() : fragmentSettings->ModifiedProfiles();
                if (guid != winrt::guid{})
                {
                    if (applyToSettings)
                    {
                        _appendProfile(std::move(profile), guid, settings);
                    }
                    destinationSet.Append(winrt::make<FragmentProfileEntry>(guid, hstring{ til::u8u16(Json::writeString(styledWriter, profileJson)) }));
                }
```

No point in introducing more common subexpressions if we don't have to!

I could not delete the below code with my suggestion, so take this as a deletion too

Reply:
this also keeps the layout of the original code more intact, and avoids reading `guid` until we have actually checked `HasGuid` :P